spec/linter-registry-spec.js

```diff
@@ -262,14 +263,97 @@ describe('LinterRegistry', function() {
       const linter = getLinter()
       const editor = atom.workspace.getActiveTextEditor()
       linterRegistry.addLinter(linter)
-      linter.lint = function() { return false }
-      const promise = linterRegistry.lint({ editor, onChange: false })
+      linter.lint = function() {
+        if (timesBegan === 2) {
+          return null
+        }
+        return []
+      }
+
+      promise = linterRegistry.lint({ editor, onChange: false })
+      expect(timesBegan).toBe(1)
+      expect(timesUpdated).toBe(0)
+      expect(timesFinished).toBe(0)
+      expect(await promise).toBe(true)
+      expect(timesUpdated).toBe(1)
+      expect(timesFinished).toBe(1)
+      promise = linterRegistry.lint({ editor, onChange: false })
+      expect(timesBegan).toBe(2)
+      expect(timesUpdated).toBe(1)
+      expect(timesFinished).toBe(1)
+      expect(await promise).toBe(true)
+      expect(timesUpdated).toBe(1)
+      expect(timesFinished).toBe(2)
+    })
+    it('cries if response is not array and is non-null', async function() {
```

Comment:
Maybe `'cries if response is not an Array or null'`? The current wording works, but is a bit awkward.